extra_geom/geom_from_karabo.py

```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+import argparse
+import asyncio
+import base64
+import gzip
+import pickle
+
+from karabo_proxy import AsyncKaraboProxy
+
+FRM_DIMS = {
+    "crystfel": ('frame', 'modno', 'ss', 'fs'),
+    "cheetah": ('frame', 'ss', 'fs'),
+}
+
+presets = {
+    "SPB/AGIPD1M": {
+        "proxy_url": "http://spb-rr-sys-con-gui1:8484",
+        "geometry_device": "SPB_DET_AGIPD1M-1/CAL/GEOMETRY",
+        "z_motor": "SPB_IRU_AGIPD1M/MOTOR/Z_STEPPER",
+        "energy_device": "SPB_XTD2_UND/DOOCS/ENERGY",
+        "clen_min": 0.119,
+    },
+    "SPB/JF4M": {
+        "proxy_url": "http://spb-rr-sys-con-gui1:8484",
+        "geometry_device": "SPB_IRDA_JF4M/CAL/GEOMETRY",
+        "z_motor": "SPB_IRDA_JF4M/MOTOR/Z",
+        "energy_device": "SPB_XTD2_UND/DOOCS/ENERGY",
+        "clen_min": 0.020,
+    },
+    "MID": {
+        "proxy_url": "http://mid-rr-sys-con-gui1:8484",
+        "geometry_device": "MID_DET_AGIPD1M-1/CAL/GEOMETRY",
+        "z_motor": "MID_EXP_UPP/MOTOR/T4",
+        "energy_device": "MID_XTD1_UND/DOOCS/ENERGY",
+        "clen_min": 0.0,
+    },
+}
+
+
+def deserialize_geometry(serialized: str):
+    return pickle.loads(gzip.decompress(base64.b64decode(serialized.encode())))
```

Comment:
I just realised what this is doing.

Unpickling may not work if the version of EXtra-geom installed in Karabo is different from the version locally. Recent versions haven't changed the structure much, but that's not guaranteed for the future, and the failures if you unpickle something from a different version can be very confusing. It also means that anyone calling this is potentially vulnerable to a MITM attack: unpickling is equivalent to `eval()` in security terms.

So I'd rather not do this if we can avoid it. If a major use case is to retrieve a `.geom` file, one option is that the geometry device could expose a slot to create the file, and we call that and write the returned data directly to a file.